### PR TITLE
LPS-132754

### PR DESCRIPTION
**Pull request by martamedio**
Hey @stian-sigvartsen, could you take a look to this changes? 

This is what I have found about this issue:
- First we execute [this process](https://github.com/liferay-appsec/liferay-portal/blob/master/modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/SamlServiceUpgrade.java#L123), where we select some fields to move their value into `SamlPeerBinding` and then [we drop those columns](https://github.com/liferay-appsec/liferay-portal/blob/master/modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/SamlServiceUpgrade.java#L130) from `SamlIdpSpSession`: I mention this because yesterday we were surprised that we selected columns that we thought did not exist.
- ~Note I have changed how we calculate the final value of `samlPeerBindingId`, [I've extracted it to Java](https://github.com/liferay-appsec/liferay-portal/commit/d0a7ab1f66fa2c930953c478580fe0cfb0805203), so please check that I'm really doing what's expected.~
- I think the duplicate `group by` field is causing problems on some databases
- I think [this lowercase](https://github.com/liferay-appsec/liferay-portal/commit/4052c6e05f55e5ba97f97d4d7379127595b6fbed) change is mandatory for some databases

Thanks!

### CODE REVIEW

**Comment by martamedio** [comment]
ci:test:sf

File: modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlIdpSpSessionUpgradeProcess.java [hunk 1]
```diff
@@ -38,35 +38,40 @@ protected void doUpgrade() throws Exception {
...
 					"samlNameIdSpProvidedId, samlNameIdValue, ",
-					"samlPeerEntityId) select min(samlIdpSpSessionId) + ",
-					-samlIdpSpSessionIdOffset + latestSamlPeerBindingId,
-					", companyId, min(createDate), userId, userName, '0' as ",
-					"deleted, nameIdFormat, null as nameIdNameQualifier, null ",
-					"as nameIdSpProvidedId, nameIdValue, samlSpEntityId from ",
-					"SamlIdpSpSession group by companyId, userId, userName, ",
-					"samlSpEntityId, nameIdFormat, nameIdValue, ",
-					"samlSpEntityId"));
+					"samlPeerEntityId) select ", samlPeerBindingId,
```
Comment (stian-sigvartsen) [inline modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlIdpSpSessionUpgradeProcess.java:70]: We can't use the same SamlPeerBindingId to instead multiple rows. The original code included `min(samlIdpSpSessionId)` which gets the lowest ID for each record created by the "group by".
If we need to do this in Java then we need to do this in a loop.

Comment (martamedio) [reply modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlIdpSpSessionUpgradeProcess.java]: Understood, you're right, I'm reverting this as we discussed on Slack

**Comment by martamedio** [comment]
Please, see: https://github.com/brianchandotcom/liferay-portal/pull/102389

**Pull request closed without merging**